apps/backend/src/shared/database/supabase-query-helpers.ts

```
@@ -0,0 +1,228 @@
+			throw new BadRequestException(error.message)
+
+		case 'PGRST204': // Range not satisfiable
+			throw new BadRequestException('Invalid pagination range')
+	}
+
+	// PostgreSQL constraint violations (23xxx)
+	if (error.code?.startsWith('23')) {
+		switch (error.code) {
+			case '23505': // unique_violation
+				// Extract constraint name from error details for better messages
+				const constraintMatch = error.message.match(/constraint "([^"]+)"/)
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)